packages/lexical-list/src/formatList.ts

```diff
@@ -548,3 +550,25 @@ export function $handleListInsertParagraph(): boolean {
 
   return true;
 }
+
+/**
+ * Handles the command to update the start number of an ordered list.
+ * It retrieves the ListNode by its key, sets the new start number,
+ * and updates the values of its child ListItemNodes.
+ * @param editor - The LexicalEditor instance.
+ * @param listNodeKey - The key of the ListNode to update.
+ * @param newStart - The new start number for the list.
+ */
+export function $handleUpdateListStart(
+  editor: LexicalEditor,
+  listNodeKey: NodeKey,
+  newStart: number,
+): void {
+  editor.update(() => {
+    const listNode = $getNodeByKey(listNodeKey);
+    if ($isListNode(listNode) && listNode.getListType() === 'number') {
+      listNode.setStart(newStart);
+      updateChildrenListItemValue(listNode);
+    }
+  });
+}
```

Comment:
Using editor here is unnecessary and actively error-prone (nesting editor.update calls has very unexpected behavior). Other functions with this sort of API have already been deprecated. The command listener already has an implicit editor.update and can directly call dollar functions.

```suggestion
/**
 * Handles the command to update the start number of an ordered list.
 * It retrieves the ListNode by its key, sets the new start number,
 * and updates the values of its child ListItemNodes.
 * @param editor - The LexicalEditor instance.
 * @param listNodeKey - The key of the ListNode to update.
 * @param newStart - The new start number for the list.
 */
export function $handleUpdateListStart(
  listNodeKey: NodeKey,
  newStart: number,
): void {
  const listNode = $getNodeByKey(listNodeKey);
  if ($isListNode(listNode) && listNode.getListType() === 'number') {
    listNode.setStart(newStart);
    updateChildrenListItemValue(listNode);
  }
}
```